utils.sh

```diff
@@ -269,7 +269,11 @@ function sync_repo_and_patch {
 function generate_auth_template {
     # clouds.yaml
     OCP_VERSIONS_NOAUTH="4.3 4.4 4.5"
-    if [[ "$OCP_VERSIONS_NOAUTH" == *"$OPENSHIFT_VERSION"* ]]; then
+
+    # Fix the version if OPENSHIFT_VERSION follows x.y.z format to change it to x.y
+    OPENSHIFT_VERSION_TRIM=$(echo "$OPENSHIFT_VERSION" | grep -o '^[[:digit:]].[[:digit:]]')
```

Comment:
Sorry, my suggestion earlier was incomplete - I forgot to escape the `.` so this will match e.g 4.3 and 4x3, also it won't work when either number reaches double figures, so should probably be:

```
grep -oE '^[[:digit:]]+\.[[:digit:]]+'
```

Reply:
Modified. Aslo added error handing when version is empty string (grep will throw error).